api4/user.go

```diff
@@ -84,6 +85,67 @@ func getUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getUsers(c *Context, w http.ResponseWriter, r *http.Request) {
```

Comment:
It does do a lot and it would be a bit nicer to have the permissions split out by route, but I don't like having to create new endpoints for all the different ways of selecting users and doing it with query parameters makes it easier to combine different parameters for different results. I'm 3/5 on keeping it all in the same route

I could split it out into more functions so it's more readable/maintinable and the main `getUsers` function would just use the correct function based on the query parameters, if you think that would be better.